armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -648,6 +653,64 @@ def _getNucTempHelper(self):
         return nvt, nv
 
 
+class CylindricalComponentsDuctHetAverageBlockCollection(
```

Comment:
Honest question: why is this class in ARMI?

Is it just because `BLOCK_COLLECTIONS` is used in `blockCollectionFactory()`?

Reply:
Yes, pretty much. It's just an extension of the `CylindricalComponentsAverageBlockCollection` with modified behavior in a few circumstances.